app/scripts/modules/core/src/application/application.model.ts

```diff
@@ -79,9 +79,9 @@ export class Application {
    * Which data source is the active state
    * @type {ApplicationDataSource}
    */
-  public activeState: ApplicationDataSource = null;
+  public activeState: ApplicationDataSource<any> = null;
 
-  public activeDataSource$ = new ReplaySubject<ApplicationDataSource>(1);
+  public activeDataSource$ = new ReplaySubject<ApplicationDataSource<any>>(1);
```

Comment:
If you give the generic parameter a default value, you don't have to do this `<any>` thing all over the place.  

```
interface ApplicationDataSource<T = any> {```

Reply:
yeah... I know and good call. I had originally left out a default as a convenient way of having the TS compiler give me a hit-list of spots the type was used so I could see whether any of them were capable of getting narrowed down beyond `any` (a few were). Then, Thursday Erik had some sort of half-baked idealogical thing going on where he thought forcing people to specify a type might encourage more specific typing in the future rather than throwing `any` around because it's the default, which led to keeping it in this state.

Monday Erik thinks that was maybe a little too ideological, and I'm going to just default it.